### PR TITLE
WIP(convert tool to transform)-135 add DICOM loader

### LINKED ISSUE #135: add a DICOM loader
Nibabal is used to load Nifti files,
would be great to have preliminary support of loading DICOM (using pydicom)

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #135 .

### Description
This PR implemented a medical image format converter.
Convert medical image formats with option to resample image.
    Supported input formats: _.dcm, .nii, .nii.gz, .mha, .mhd._
    Supported output formats: _.nii, .nii.gz, .mha, .mhd._

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

**Comment by aylward** [comment]
I don't recommend creating dependencies on SimpleITK.   Consider using the full ITK - pip install itk

File: monai/utils/medical_image_converter.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+import numpy as np
+import SimpleITK as Sitk
```
[the remark this answers is not in the document]
Comment (madil90) [reply monai/utils/medical_image_converter.py]: @Nic-Ma Why not use pydicom instead? It's simpler to use. I have some code I can submit tomorrow.

[the remark this answers is not in the document]
Comment (aylward) [reply monai/utils/medical_image_converter.py]: PyDicom is a great idea for dicom support!

However, I don't think pydicom should be used "instead" of ITK, but perhaps it should be used in addition to ITK.  ITK can provide support for 50+ additional file formats.

[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/utils/medical_image_converter.py]: Hi @madil90 ,

Please feel free to add pydicom as another option or change my current PR directly.
Thanks.

[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/utils/medical_image_converter.py]: > PyDicom is a great idea for dicom support!
> 
> However, I don't think pydicom should be used "instead" of ITK, but perhaps it should be used in addition to ITK. ITK can provide support for 50+ additional file formats.

Hi Stephen,

Thanks very much for your suggestions!
According to internal discussion with @yanchengnv @wyli, we aligned that:
For the Alpha release before GTC San Jose, we will only provide a primary DICOM converter.
I think we will make a powerful tool after GTC.
Thanks.

**Comment by Nic-Ma** [comment]
Hi All,

I updated the code to be compatible with master.
Could you please help review it again?
Thanks.

**Comment by wyli** [comment]
i think we need to have a loader wrapped as a transform, to show the idea that monai doesn't implement its own but build wrapper around external packages for this type of IO tasks. provide a CLI script is not proper integration... as a starting point let's use pydicom

**Comment by zsdqui** [comment]
any update on this WIP? Any thoughts on adding support for dicom-RT. Being able to load RT-struct files would greatly assist utilizing MONAI for radiotherapy applications.

**Comment by Nic-Ma** [comment]
> any update on this WIP? Any thoughts on adding support for dicom-RT. Being able to load RT-struct files would greatly assist utilizing MONAI for radiotherapy applications.

Hi @zsdqui ,

Thanks for your suggestion.
I think our working group is discussing the DICOM support.
Thanks.

**Comment by wyli** [comment]
this is out-of-date and becomes really confusing, I'm closing it now (the branch will be retained)

**Pull request closed without merging**